csharp/ql/src/semmle/code/csharp/commons/Util.qll

```diff
@@ -13,3 +13,13 @@ class MainMethod extends Method {
     else this.getNumberOfParameters() = 0
   }
 }
+
+class RecordCloneMethod extends Method {
+  RecordCloneMethod() { exists(WithExpr we | we.getTarget() = this) }
```

Comment:
This is only a subset of the `<Clone>$` methods. In the base PR, we already look up these methods, let's use that query to decide what is a clone method.